### DISCUSSION TITLE
Output SARIF format is confusing

### DISCUSSION THREAD

**Discussion by yuligesec** [projectdiscovery/nuclei#2607] [Q&A] [answered]
```  
 -se, -sarif-export string     file to export results in SARIF format
```
Although nuclei supports exporting to sarif format, it does not conform to the rules of sarif format, and the output result should be placed in the `results` field. But not.

It is confusing, like this.
```
...
"results": [
        {
          "ruleId": "tech-detect-95410662745f9d47610c389af7d617774a98aa0c",
          "ruleIndex": 0,
          "level": "note",
          "message": {
            "text": "http://127.0.0.1:18080"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "README.md"
                },
                "region": {
                  "startLine": 1,
                  "startColumn": 1,
                  "endLine": 1,
                  "endColumn": 1
                }
              },
              "message": {
                "text": "http://127.0.0.1:18080"
              }
            }
          ]
        },
...
```
Sarif Viewer can't view it and get any useful information! 
![image](https://user-images.githubusercontent.com/32768214/190394145-1f3068fe-e9bb-45e4-8159-8b1b71ea20bb.png)

**Comment by olearycrew** [comment]
@yuligesec can you provide a side by side example of what you're expecting vs. what Nuclei is outputting? I'm not quite following what the diff should be here.

**Comment by tarunKoyalwar** [comment, marked as answer]
@yuligesec , this was resolved with https://github.com/projectdiscovery/nuclei/pull/2925 . but there is a known bug https://github.com/projectdiscovery/nuclei/issues/3221
